File: Medium/Java/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int longest = 0;
        HashSet<Character> set = new HashSet();
        
        // Go over the string, comparing each character
        // to the next. If a character is new, place it
        // into the HashSet. Once a repeat is found,
        // the size of the HashSet is the length of substring
        for(int i = 0; i < s.length(); i++){
            for(int j = i; j < s.length(); j++){
                if(set.contains(s.charAt(j))){
                    j = s.length();
                }else{
                    set.add(s.charAt(j));
                }
            }
            longest = Math.max(set.size(), longest);
            set.clear();
        }
        
        return longest;
    }
}

File: Medium/JavaScript/3. Longest Substring Without Repeating Characters.js
var lengthOfLongestSubstring = function(s) {
    let longest = 0;
    let word = new Set();

    // Go over the string, comparing each character
    // to the next. If a character is new, place it
    // into the Set. Once a repeat is found,
    // the size of the Set is the length of the substring
    for(let i = 0; i < s.length; i++){
        for(let j = i; j < s.length; j++){
            if(word.has(s.charAt(j))){
                j = s.length;
            }else{
                word.add(s.charAt(j));
            }
        }
        longest = Math.max(word.size, longest);
        word.clear();
    }

    return longest;
};
